firstapp/app/src/androidTest/java/com/example/firstapp/UnitTests.java

```diff
@@ -0,0 +1,61 @@
+package com.example.firstapp;
+import androidx.test.platform.app.InstrumentationRegistry;
+import androidx.test.uiautomator.By;
+import androidx.test.uiautomator.UiDevice;
+import androidx.test.uiautomator.UiObject;
+import androidx.test.uiautomator.UiObject2;
+import androidx.test.uiautomator.UiObjectNotFoundException;
+import androidx.test.uiautomator.UiSelector;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import androidx.test.uiautomator.Until;
+
+import org.junit.Test;
+
+import java.io.IOException;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.prefs.BackingStoreException;
+
+import static org.junit.Assert.*;
+
+/**
+ * Example local unit test, which will execute on the development machine (host).
+ *
+ * @see <a href="http://d.android.com/tools/testing">Testing documentation</a>
+ */
+public class UnitTests {
+    private static final String BASIC_SAMPLE_PACKAGE = "com.example.firstapp";
+    private UiDevice device;
+    private static final int TIMEOUT = 6000;
+
+
+    @Test
+    public void launching () throws InterruptedException, IOException {
```

Comment:
Generally no space before open parenthesis of method names:

    launching () -> launching()